internal/server/edit_my_details_test.go

```diff
@@ -0,0 +1,205 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/ministryofjustice/opg-sirius-user-management/internal/sirius"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockEditMyDetailsClient struct {
+	count       int
+	lastCookies []*http.Cookie
+	lastRequest string
+	err         error
+	errSave     error
+	data        sirius.MyDetails
+}
+
+func (m *mockEditMyDetailsClient) MyDetails(ctx context.Context, cookies []*http.Cookie) (sirius.MyDetails, error) {
+	m.count += 1
+	m.lastCookies = cookies
+	m.lastRequest = "MyDetails"
+
+	return m.data, m.err
+}
+
+func (m *mockEditMyDetailsClient) EditMyDetails(ctx context.Context, cookies []*http.Cookie, id int, phoneNumber string) error {
+	m.count += 1
+	m.lastCookies = cookies
+	m.lastRequest = "EditMyDetails"
```

Comment:
Do you not want to check it was called with the expected args too? It might also make more sense to have a `count` and an `editCount` so you can tell the calls apart instead of `lastRequest`. Or maybe we should use https://godoc.org/github.com/stretchr/testify/mock